### PR TITLE
Update to JasperReports 5.6.1

### PR DESCRIPTION
**Pull request by AndreHeinecke**
Hi,

These commits update jasperreports to 5.6.1. Jasperreports 5.6.0 introduced a new subreport attribute (overflowType) which can be used to control the overflow behaviour of legend subreports. Where Previously a Large Legend subreport would stretch over other content you now have better control about where large legends land (e.g. another column).

As Jasperreports deprecated the JRExporterParameter API in a way that broke the PDF Exporter in mapfish-print (they changed types on PDF_VERSION) and avoid usage of deprecated functions I've ported mapfish-print to use the new Parameter API.
Basically instead of using a long list of Parameters each exporter now has an ExporterInput, ExporterOutput and ExporterConfiguration object to hold the same information.

The net.sf.jasperreports.repo.RepositoryContext and related methods are now removed in Jasperreports. So I also had to remove those Overriden methods in MapfishPrintRepositoryService.java too.

Regards,
Andre

### CODE REVIEW

**Comment by jesseeichar** [comment]
Very cool, I will properly review this tomorrow or later today.

**Comment by tsauerwein** [comment]
This is looking good to me. Thanks for the contribution, Andre!

**Comment by sbrunner** [comment]
To me to :-)
